File: app/models/recipe.rb
class Recipe
  include Mongoid::Document
  include Mongoid::Timestamps
  include Mongoid::Slug

  field :name, type: String
  field :description, type: String
  field :photo, type: String
  field :preptime, type: String
  field :cooktime, type: String
  field :servings, type: String
  field :difficulty, type: String
  field :cuisines, type: Set

  slug :name, history: true

  embeds_many :steps
end

File: Core/Recipe.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace InspiredCooking.Core
{
    public partial class Recipe
    {

        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }
        public string Photo { get; set; }

        [Required]
        public string PrepTime { get; set; }

        [Required]
        public string CookTime { get; set; }

        [Required]
        public int Servings { get; set; }

        [Required]
        public DifficultyType Difficulty { get; set; }

        [Required]
        public CuisineType Cuisine { get; set; }

        public List<Ingredient> Ingredients { get; set; } = new List<Ingredient>();
        public List<Step> Steps { get; set; } = new List<Step>();

        public ApplicationUser User { get; set; }

        public string UserId { get; set; }

        public IEnumerable<FavoriteRecipe> FavoritedByUsers { get; set; }

        public Image Image { get; set; }
        public int? ImageId { get; set; }
    }
}